### ISSUE TITLE
Regression since 1.17.0 - policy does not recognize my github app name

### ISSUE THREAD

**Issue by NargiT** [palantir/policy-bot#230]
Hello,

We are using GitHub Enterprise Server 2.20.5 so I cannot do any test on +1.17.1.

I did an upgrade from 1.16.3 to 1.17.0 and to 1.17.1 and now policy is not able to recognize the bot user and fail the status check.

Some context:
I use a Github App to create pull request to update my repository (same idea as renovate or dependabot).
The commit and the author have the following format:

```
Author:     mybot[bot] <mybot[bot]@users.noreply.github.domain.ch>
AuthorDate: Tue Sep 8 06:39:34 2020 +0000
Commit:     mybot[bot] <mybot[bot]@users.noreply.github.domain.ch>
CommitDate: Tue Sep 8 06:39:34 2020 +0000
```

```
policy:
  approval:
    - mybot upgrade

approval_rules:
  - name: mybot upgrade
    if:
      has_author_in:
        users: ['mybot[bot]']
      author_is_only_contributor: true
      targets_branch:
        pattern: ^master
    options:
      invalidate_on_push: true
```
The error

```
The pull request author "mybot" does not meet the required membership conditions
```

And if I modify the users list to `['mybot']` only I get another error

```
Commit 8a82a72953 was authored or committed by a different user
```

I also tried to remove the "bot" and only use mybot for commit and pull request creation, but end up with the previous error `Commit xxxx was authored or committed by a different user`

This is working very well with 1.16.3 but since 1.17.0 policy does not recognize my bot anymore.
Which field is used to match the user name ?

Looks like the issue comes from this changes https://github.com/palantir/policy-bot/compare/v1.16.3...v1.17.0

**Comment by bluekeyes** [comment]
The first message

> The pull request author "mybot" does not meet the required membership conditions

uses the `user.login` field of the [pull request](https://docs.github.com/en/rest/reference/pulls#get-a-pull-request) or the `author.login` field of the [`PullRequest`](https://docs.github.com/en/graphql/reference/objects#pullrequest) GraphQL object. I think there actually is a bug here, as the GraphQL API returns logins in a different format and we didn't account for that when getting the pull request author. However, that bug was introduced back in 1.8.0.

The second message

> Commit 8a82a72953 was authored or committed by a different user

uses the `author.login` and `committer.login` fields of the [`Commit`](https://docs.github.com/en/graphql/reference/objects#commit) GraphQL object. In this case, we properly account for the difference between the v3 and v4 APIs. The values are compared against the name of the user who opened the PR, which is subject to the bug I described earlier.

Since very little changed between 1.16.3 and 1.17.0, I think this must be related to the `go-github` library upgrade. Maybe something now triggers the bug in computing the pull request author when it was avoided before.

**Comment by NargiT** [comment]
Thanks for your work, if you can backport this fix to 1.17.x I would be able to validate that it works.

**Comment by bluekeyes** [comment]
I released 1.17.2 including this fix. Hopefully it solves the problem for you.

**Comment by NargiT** [comment]
Perfect it works! Thanks !!!
